src/core/components/InstallButton/index.js

```diff
@@ -22,29 +24,49 @@ export class InstallButtonBase extends React.Component {
     status: PropTypes.string.isRequired,
   }
 
+  static defaultProps = {
+    clientSupportsAddons: _clientSupportsAddons,
+  }
+
   installTheme = (event) => {
     event.preventDefault();
     const { addon, status, installTheme } = this.props;
     installTheme(event.currentTarget, { ...addon, status });
   }
 
   render() {
-    const { addon, className, hasAddonManager, i18n, size } = this.props;
+    const {
+      addon,
+      clientSupportsAddons,
+      className,
+      hasAddonManager,
+      i18n,
+      size,
+    } = this.props;
     const useButton = hasAddonManager !== undefined && !hasAddonManager;
     let button;
+    const buttonIsDisabled = !clientSupportsAddons();
+    const buttonClass = classNames('InstallButton-button', {
+      'InstallButton-button--disabled': buttonIsDisabled,
+    });
+
     if (addon.type === ADDON_TYPE_THEME) {
       button = (
         <Button
+          disabled={buttonIsDisabled}
           data-browsertheme={JSON.stringify(getThemeData(addon))}
           onClick={this.installTheme}
           size={size}
-          className="InstallButton-button">
+          className={buttonClass}>
           {i18n.gettext('Install Theme')}
         </Button>
       );
     } else {
       button = (
-        <Button href={addon.installURL} size={size} className="InstallButton-button">
+        <Button
+          href={buttonIsDisabled ? null : addon.installURL}
```

Comment:
Maybe we should file a bug to show a link to the add-on XPI directly if the install button is disabled. Could be handy for debugging but also as a failsafe if our detection went wrong or something.